src/App.tsx

```diff
@@ -1,39 +1,58 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+import { getRandomName } from './function';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
 
-  return `Clock-${value}`;
-}
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  clockNameId = 0;
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleLeftClick);
+    document.addEventListener('contextmenu', this.handleRigthClick);
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+    this.clockNameId = window.setInterval(this.clockNameChanger, 3300);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleLeftClick);
+    document.removeEventListener('contextmenu', this.handleRigthClick);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  clockNameChanger = () => {
```

Comment:
```suggestion
  changeClockName = () => {
```
![image](https://user-images.githubusercontent.com/96409563/229709315-1913bb99-0a27-4e3d-bbe2-94d8a855a5f6.png)